### DISCUSSION TITLE
Add an option to display emotes in the Split Input

### DISCUSSION THREAD

**Discussion by RandomNerd1** [Chatterino/chatterino2#4953] [Ideas] [answered]
It'd be nice to have a 'complete' preview of what I'm about to send into the chat, but right now, any emotes in the area where you're typing, whether keyed in manually or tab-completed, just show up as the characters that comprise the emote, instead of the emote itself.

**Comment by Mm2PL** [comment]
Making a good What-You-See-Is-What-You-Get chat input is not easy. Especially with Twitch as you can't tell what emote the current user has because Twitch just doesn't give that information easily. 
Seemingly none of the main contributors of the project seem to wanted to make this feature over the project's existence up to this point. I am not saying no to this suggestion however I believe someone implementing this quick is unlikely.

**Reply by Felanbird** [reply]
ye it would be cool to have but a huge pain in the ass to implement

**Reply by RandomNerd1** [reply]
Thanks for the quick reply!

Hm...I can understand what you're getting at when it comes to emotes fully typed out manually, but at least when it comes to tab-completion, if you've just tab-completed a given emote, shouldn't Chatterino know which it is, and be able to 'confidently' display it accordingly in the chat box? 🤔

**Reply by Mm2PL** [reply, marked as answer]
The annoying part is the showing images in the input and tracking where images should be replaced by text on message send. Finding an emote in the message isn't that hard.

> if you've just tab-completed a given emote, shouldn't Chatterino know which it is, and be able to 'confidently' display it accordingly in the chat box? 🤔

Yeah, for most emotes it would just work especially if they were completable. Some emotes like ones you pick for bits unlocks, event ones and hypetrain ones are another level of pain. Chatterino just can't know about them. Anything thirdparty would just work.

These are the currently uncompletable emote sets I have:
![image](https://github.com/Chatterino/chatterino2/assets/25011746/2f8396e5-d0a6-4d50-9dcd-32012ef4c729)

 This impacts chat completion but it's an even bigger problem with the client-side logic needed for this feature. When sending messages this isn't an issue because Twitch's chat servers give us all information to show emotes when receiving the just sent message.

**Reply by Nerixyz** [reply]
To add to this:

Rendering animated images in a text input isn't trivial in Qt. Only static image resources are supported. You'd basically need to reinvent [`QTextEdit`](https://doc.qt.io/qt-6/qtextedit.html) and [`QTextDocument`](https://doc.qt.io/qt-6/qtextdocument.html) and thus hand roll your own text layout (again) and text interaction (might have platform specific features), which _in my opinion_ isn't worth it for this feature. So you could only see static images when typing emotes.

**Reply by RandomNerd1** [reply]
Ouch. Unfortunate, but understandable. Might as well close this, then. Thanks in any case, for the explanation :)

**Comment by RandomNerd1** [comment]
> To add to this:
> 
> Rendering animated images in a text input isn't trivial in Qt. Only static image resources are supported. You'd basically need to reinvent [`QTextEdit`](https://doc.qt.io/qt-6/qtextedit.html) and [`QTextDocument`](https://doc.qt.io/qt-6/qtextdocument.html) and thus hand roll your own text layout (again) and text interaction (might have platform specific features), which _in my opinion_ isn't worth it for this feature. So you could only see static images when typing emotes.
